enclave_manager/setup_enclave_attestation.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+
+# Copyright 2018-2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import subprocess
+
+# This should only be run with python3
+import sys
+if sys.version_info[0] < 3:
+    print('ERROR: must run with python3')
+    sys.exit(1)
+
+from setuptools import setup, find_packages, Extension
+
+tcf_root_dir = os.environ.get('TCF_HOME', '../')
+
+enclave_bridge_wrapper_path = os.path.join(tcf_root_dir,
+         'tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge_wrapper')
+enclave_bridge_path = os.path.join(tcf_root_dir,
+         'tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge')
+src_swig_file_name = os.path.join(tcf_root_dir,
+         'enclave_manager/avalon_enclave_manager/attestation/enclave_info.i')
+
+# If enclave_type is not set then default to singleton
+enclave_type =  os.environ.get('ENCLAVE_TYPE', 'SINGLETON').upper()
+
+## -----------------------------------------------------------------
+## Set up the enclave
+## -----------------------------------------------------------------
+debug_flag = os.environ.get('TCF_DEBUG_BUILD',0)
+
+compile_args = [
+    '-std=c++11',
+    '-Wno-switch',
+    '-Wno-unused-function',
+    '-Wno-unused-variable',
+]
+
+# By default the extension class adds '-O2' to the compile
+# flags, this lets us override since these are appended to
+# the compilation switches.
+if debug_flag :
+    compile_args += ['-g']
+
+include_dirs = [
+    enclave_bridge_wrapper_path,
+    enclave_bridge_path,
+    os.path.join(tcf_root_dir, 'common/cpp'),
+    os.path.join(tcf_root_dir, 'tc/sgx/trusted_worker_manager/common'),
+    os.path.join(tcf_root_dir, 'common/cpp/crypto'),
+    os.path.join(tcf_root_dir, 'common/cpp/packages/base64'),
+]
+
+library_dirs = [
+    os.path.join(enclave_bridge_path, 'build/lib'),
+    os.path.join(tcf_root_dir, "common/cpp/build"),
+]
+
+libraries = []
+
+if enclave_type == "KME":
+    module_name = 'enclave_info_kme'
+    libraries.append('avalon-kme-enclave-bridge')
+elif enclave_type == "WPE":
+    module_name = 'enclave_info_wpe'
+    libraries.append('avalon-wpe-enclave-bridge')
+elif enclave_type == "SINGLETON":
+    module_name = 'enclave_info_singleton'
+    libraries.append('avalon-singleton-enclave-bridge')
+else:
+    print('Unsupported enclave type passed in the config %s', enclave_type)
+    sys.exit(1)
+
+lib_name = 'avalon_enclave_manager.attestation._' + module_name
```

Comment:
My understating is we can use the same library name for all enclave types. Since the wheel file generated is different for each enclave type and the library is contained within the wheel package, there shouldn't be any conflict. Did you try using same library name and different wheel  package ?

Reply:
It is a swig generated shared library name, it has to different for kme, wpe and singlteotn.